pkg/metrics/metrics.jsx

```diff
@@ -905,13 +905,27 @@ const PCPConfig = ({ buttonVariant, firewalldRequest }) => {
         redis_name = "redis.service";
     }
 
-    debug("PCPConfig s_pmlogger.state", s_pmlogger.state);
+    debug("PCPConfig s_pmlogger.state", s_pmlogger.state, "needs logout", needsLogout);
     debug("PCPConfig s_pmproxy state", s_pmproxy.state, "redis exists", s_redis.exists, "state", s_redis.state, "redis-server exists", s_redis_server.exists, "state", s_redis_server.state);
 
     if (!superuser.allowed)
         return null;
 
     const handleSave = () => {
+        // when enabling pmlogger, install cockpit-pcp/pcp on demand
+        if (dialogLoggerValue && s_pmlogger.state === undefined && !needsLogout) {
+            debug("PCPConfig: request to enable pmlogger, but missing pmlogger.servicie, offering install; needs logout", needsLogout);
+            setDialogVisible(false);
+            install_dialog("cockpit-pcp")
+                    .then(() => {
+                        debug("PCPConfig: cockpit-pcp installation successsful");
+                        setNeedsLogout(true);
+                        handleSave();
+                    })
+                    .catch(() => null); // ignore cancel
```

Comment:
This will however ignore all exceptions not just cancelled promise, right?

Reply:
Right, but the dialog already handles package install failures etc. internally (as inline alerts). `install_dialog()` itself is documented accordingly:

> The install_dialog function returns a promise that is fulfilled when the dialog closes after
> a successful installation.  The promise is rejected when the user cancels the dialog.

We use it the same way in pkg/packagekit/autoupdates.jsx or pkg/storaged/optional-panel.jsx.

But indeed I just saw that we don't handle cancel() in handleInstall() on the metrics page -- this causes an unhandled exception that appears in the log. I'll fix that in a separate commit.

Reply:
ack thanks